syntax = "proto3";

package entity;

// Entity 消息定义
message Entity {
  string name = 1;                // 对应 String(256), 主键
  repeated string join_keys = 2;  // 对应 JSON List
}

// 创建 Entity 的请求消息
message CreateEntityRequest {
  Entity entity = 1;
}

// 创建 Entity 的响应消息
message CreateEntityResponse {
  bool success = 1;
  string message = 2;
  Entity entity = 3;
}

// 查询 Entity 的请求消息
message GetEntityRequest {
  string name = 1;
}

// 查询 Entity 的响应消息
message GetEntityResponse {
  Entity entity = 1;
  bool found = 2;
  string message = 3;
}

// Entity 服务定义
service EntityService {
  // 创建新 Entity
  rpc CreateEntity (CreateEntityRequest) returns (CreateEntityResponse);
  
  // 查询 Entity
  rpc GetEntity (GetEntityRequest) returns (GetEntityResponse);
}
